test/sand/erc20BasicApproveExtension.test.ts

```diff
@@ -37,6 +37,77 @@ async function testFunction(
 }
 
 describe('ERC20BasicApproveExtension', function () {
+  it('ApproveAndCall calling buyLandWithSand', async function () {
+    const {
+      landContract,
+      lands,
+      landSaleContract,
+      sandContractAsUser0,
+      sandContract,
+      user0,
+    } = await setupERC20BasicApproveExtension();
+    const emptyReferral = '0x';
+    const secret =
+      '0x4467363716526536535425451427798982881775318563547751090997863683';
+    const saltedLands = MerkleTreeHelper.saltLands(lands, secret);
+    const landHashArray = MerkleTreeHelper.createDataArray(saltedLands);
+    const land = saltedLands
+      .filter((l: SaltedSaleLandInfo) => l.size === 6)
+      .find((l: SaltedSaleLandInfo) => !l.reserved);
+
+    expect(land).to.not.be.equal(undefined);
+    if (land !== undefined) {
+      const tree = new MerkleTree(landHashArray);
+      const proof = tree.getProof(MerkleTreeHelper.calculateLandHash(land));
+      const totalSandBalance = BigNumber.from(100000).mul(
+        `1000000000000000000`
+      );
+      const approvalAmount = BigNumber.from(5000).mul(`1000000000000000000`);
+      const encodedABI = await landSaleContract.populateTransaction.buyLandWithSand(
+        user0,
+        user0,
+        zeroAddress,
+        land.x,
+        land.y,
+        land.size,
+        land.price,
+        land.price,
+        land.salt,
+        [],
+        proof,
+        emptyReferral
+      );
+      expect(encodedABI.data).to.not.be.equal(undefined);
+      if (encodedABI.data) {
+        const data = encodedABI.data.substring(2);
```

Comment:
could you not just pass the data as is ?, it should work and would make the test more clear.
I made similar comment on slack for the other tests